nservicebus/azure-service-bus/performance-tuning_body_asb_[7,).partial.md

```diff
@@ -0,0 +1,63 @@
+
+  * `LimitMessageProcessingConcurrencyTo()`: This is the global concurrency limit across all receive entities. Ensure this value matches the per receiver desired concurrency for the scenario, multiplied by the number of receivers.
+  * `MessageReceivers().PrefetchCount()`: Value is per receiver, in general the following rule applies: The lower the throughput rate of an individual receiver, the lower this value should be.
+  * `Transactions()`: The most significant performance impact comes from the transactional guarantees needed, `SendsAtomicWithReceive` will offer less performance then `ReceiveOnly`.
+
+### Sends Atomic With Receive
+
+In order to guarantee that sends occur as an atomic operation with the receive confirmation, the ASB SDK requires that both operations are enclosed in a `Serializable` transaction. This implies that:
+
+  * All atomic operations are executed in a 'Serial' fashion, i.e. 1 by 1, therefore it does not make sense to allow many concurrent operations per client, nor does it make much sense to prefetch a lot of messages.
+	  - Concurrency: Optimal values are 2 or 4 per client
+	  - PrefetchCount: Optimal value is 20
+  * Setting up the transactional connection takes up quite some time, this causes the first operational minutes to perform very 'bumpy', but after a while the throughput becomes consistent.
+  * All send operations that occur in the scope of a single receive (inside a handler), must fit into the transaction. Therefore a maximum of 100 messages can be sent this way. The good news is that doing so has little additional implications on performance tuning as 1 send operation or 100 is actually the same thing except for the size of the operation.
+
+NOTE: Send operations inside a handler are only executed when the handler operation returns, unless the send options request for immediate dispatch. Therefore using `await` on a Send inside a handler will probably not seem to impact performance negatively, this is because no send is actually occurring, yet it is still advised to return the task instead of awaiting.
+
+### Receive only
+
+This mode has a lot less constraints and will work better with high numbers for concurrency and prefetch count, which should result in a pretty high throughput (likely over 1000/s even for remote connections)
+
+  * Concurrency: Optimal values are in the range of 128 per client.
+  * PrefetchCount: Set this to 1x or 2x the per receiver concurrency, so 128 or 256.
+
+But, sending messages out at the same time does influence this mode more. If the average handler sends out an equal amount of data then it receives, then the above values still hold true. However if the handler sends out more messages (or larger messages), let's say 10x or 100x total data size per receive, then concurrency and prefetchcount should be dialed down to reduce incoming bandwidth and allow more bandwidth to the senders.
```

Comment:
it's not the number or individual size, it's the compound size. Receive 1msg of 100Kb and send 10 of 10Kb is the same as send of 1msg 100Kb